code/modules/antagonists/cult/cult_comms.dm

```diff
@@ -79,6 +79,101 @@
+	var/datum/antagonist/cult/team = owner.mind.has_antag_datum(/datum/antagonist/cult,TRUE)
+	if(world.time < cooldown)
+		to_chat(owner, span_warning("You need to wait longer before requesting assistance again!"))
+		return
+	if(team.cult_team.need_of_help) //check to make sure no one else has already requested assistance
+		to_chat(owner, span_warning("Someone else is currently requesting assistance!"))
+		return
+	var/number = tgui_alert(owner, "How many cultists do you need?",, list("0", "1", "2", "3"))
+	if(number == "0")
+		to_chat(owner, "You decide not to request help.")
+		return
+	team.cult_team.need_of_help = owner
+	for(var/datum/mind/cultist as anything in get_antag_minds(/datum/antagonist/cult))
+		if(cultist.current && cultist.current.stat != DEAD && cultist.current.client)
+			var/area/location = get_area(owner)
+			var/title = "Acolyte"
+			if(owner.mind?.has_antag_datum(/datum/antagonist/cult/master))
+				title = "Master"
+			else if(!ishuman(owner))
+				title = "Construct"
+			to_chat(cultist.current, span_cultitalic("<b>[title] [findtextEx(owner.name, owner.real_name) ? owner.name : "[owner.real_name] (as [owner.name])"] has requested the assistance of [number] cultists in \the [location.name]! <a href=?src=[REF(src)];pledge_assistance=1>(Pledge Assistance)</b>"))
+			SEND_SOUND(cultist.current, sound(pick('sound/hallucinations/over_here2.ogg','sound/hallucinations/over_here3.ogg'),0,1,75))
+	to_chat(owner, span_boldnotice("You have requested assistance! Considering making a teleport rune with your dagger to help cultists reach you easier."))
+	cooldown = world.time + base_cooldown
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/reset_need_of_help,team.cult_team), base_cooldown, TIMER_STOPPABLE)
+	return
+
+///when a player clicks the "Pledge Assistance" link, they gain a tracker and the rest of cult is notified about it
+/datum/action/innate/cult/request_aid/Topic(href, href_list)
+	if(href_list["pledge_assistance"])
+		if(istype(usr, /mob/living) && IS_CULTIST(usr.mind.current))
+			var/mob/living/user = usr
+			var/datum/antagonist/cult/team = user.mind.has_antag_datum(/datum/antagonist/cult,TRUE)
+			if(!team.cult_team.need_of_help) //check to make sure someone has requested assistance
+				to_chat(user, span_warning("No one is in need of assistance right now!"))
+				return
+			if(user == team.cult_team.need_of_help)
+				to_chat(user, span_warning("You can't answer your own request for assistance!"))
+				return
+			if(user.has_status_effect(/datum/status_effect/agent_pinpointer/cult_assistance)) //if you're already helping someone, you can't pledge again
+				to_chat(user, span_warning("You are already assisting someone!"))
+				return
+			for(var/datum/mind/cultist as anything in get_antag_minds(/datum/antagonist/cult))
+				if(cultist.current && cultist.current.stat != DEAD && cultist.current.client)
+					var/title = "Acolyte"
+					if(usr.mind?.has_antag_datum(/datum/antagonist/cult/master))
```

Comment:
Change these `usr` to `user`s, which you already have a var for, but when you split this out into a proc will need anyway.